glean-core/src/metrics/memory_distribution.rs

```diff
@@ -304,4 +305,76 @@ impl MemoryDistributionMetric {
             test_get_num_recorded_errors(glean, self.meta(), error).unwrap_or(0)
         })
     }
+
+    /// Start a new histogram buffer associated with this timing distribution metric.
+    ///
+    /// A histogram buffer accumulates in-memory.
+    /// Data is recorded into the metric on drop.
+    pub fn start_buffer(&self) -> HistogramBuffer<'_> {
+        HistogramBuffer::new(self)
+    }
+
+    fn commit_histogram(&self, histogram: Histogram<Functional>) {
+        let metric = self.clone();
+        crate::launch_with_glean(move |glean| {
+            glean
+                .storage()
+                .record_with(glean, &metric.meta, move |old_value| {
+                    let mut hist = match old_value {
+                        Some(Metric::TimingDistribution(hist)) => hist,
+                        _ => Histogram::functional(LOG_BASE, BUCKETS_PER_MAGNITUDE),
+                    };
+
+                    hist.merge(&histogram);
+                    Metric::TimingDistribution(hist)
+                });
+        });
+    }
+}
+
+/// A histogram buffer associated with a specific instance of a [`MemoryDistributionMetric`].
+///
+/// Accumulation happens in-memory.
+/// Data is merged into the metric on `Drop`.
+#[derive(Debug)]
+pub struct HistogramBuffer<'a> {
+    histogram: Histogram<Functional>,
+    metric: &'a MemoryDistributionMetric,
```

Comment:
In the case of `firefox-on-glean` will `'a` be `'static`? Asked differently, how do I embed `HistogramBuffer<'a>` in a `struct` without exposing the lifetime `'a` on that struct?

Reply:
Yes, because the metric are generated as statics (well, `OnceCell`-statics), so this can get you a `LocalMemoryDistribution<'static>` out.